BitwardenShared/Core/Platform/Services/AuthenticatorSyncService.swift

```diff
@@ -319,20 +318,57 @@ actor DefaultAuthenticatorSyncService: NSObject, AuthenticatorSyncService {
     ///   - userId: The userId of the account to which the Ciphers belong.
     ///
     private func writeCiphers(ciphers: [Cipher], userId: String) async throws {
-        let account = try await stateService.getAccount(userId: userId)
-        let useKey = await vaultTimeoutService.isLocked(userId: userId)
-
         do {
-            if useKey {
-                try await authRepository.unlockVaultWithAuthenticatorVaultKey(userId: userId)
-            }
+            try await unlockVaultWithAuthenticatorVaultKey(userId: userId)
+
+            let account = try await stateService.getAccount(userId: userId)
             let items = try await decryptTOTPs(ciphers, account: account)
             try await authBridgeItemService.replaceAllItems(with: items, forUserId: userId)
+
+            try await clientService.removeClient()
         } catch {
             errorReporter.log(error: error)
+            try? await clientService.removeClient()
         }
-        if useKey {
-            await vaultTimeoutService.lockVault(userId: userId)
-        }
+    }
+
+    // MARK: Vault Unlocking
+
+    /// Initializes the SDK's crypto for any organizations the users is a member of.
+    ///
+    /// - Parameter userId: The userId the account being used to initialize the SDK.
+    ///
+    private func initializeOrganizationCrypto(userId: String) async throws {
+        let organizationKeysById = try await organizationService.fetchAllOrganizations(userId: userId)
+            .reduce(into: [String: String]()) { result, organization in
+                guard let key = organization.key else { return }
+                result[organization.id] = key
+            }
+        try await clientService.crypto().initializeOrgCrypto(
+            req: InitOrgCryptoRequest(organizationKeys: organizationKeysById)
+        )
+    }
+
+    /// Unlocks the Vault with the user's previously stored Authenticator Vault Key.
+    ///
+    /// - Parameter userId: The userId the account being used to initialize the SDK.
+    ///
+    private func unlockVaultWithAuthenticatorVaultKey(userId: String) async throws {
+        let authenticatorKey = try await keychainRepository.getAuthenticatorVaultKey(userId: userId)
+
+        let account = try await stateService.getAccount(userId: userId)
+        let encryptionKeys = try await stateService.getAccountEncryptionKeys(userId: userId)
+
+        try await clientService.crypto().initializeUserCrypto(
```

Comment:
🤔 Unless I'm missing something this won't use a separate SDK client but [reuse](https://github.com/bitwarden/ios/blob/main/BitwardenShared/Core/Platform/Services/ClientService.swift#L253) the one for the current `userId` if it exists. Did you mean to create a new SDK client for a particular user? If so, a bit of a refactor is needed to have multiple SDK clients for the same user.